datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,343 @@
+   * creating Hive external tables and BigQuery tables
+   *
+   * @param schema incoming Avro schema
+   * @return Schema Persisted Parquet schema
+   * @throws Exception
+   */
+  public static Schema generateParquetStructuredAvroSchema(Schema schema) {
+    Schema parquetAvroSchema;
+    try {
+      SchemaBuilder.FieldAssembler<Schema> fieldAssembler = SchemaBuilder.record(schema.getName())
+          .namespace(schema.getNamespace())
+          .doc(schema.getDoc())
+          .fields();
+
+      List<Schema.Field> fields = schema.getFields();
+      for (Schema.Field field : fields) {
+        Schema fieldSchema = field.schema();
+        Boolean isNotNullable = field.getObjectProp("isRequired") != null ?(Boolean) field.getObjectProp("isRequired") : false;
+        if (fieldSchema.isUnion()) {
+          for (Schema typeSchema : fieldSchema.getTypes()) {
+            if (!typeSchema.getType().getName().equalsIgnoreCase("null")) {
+              fieldAssembler = getFieldAssembler(isNotNullable, typeSchema, field, fieldAssembler);
+            }
+          }
+        } else {
+          fieldAssembler = getFieldAssembler(isNotNullable, fieldSchema, field, fieldAssembler);
+        }
+      }
+      parquetAvroSchema = fieldAssembler.endRecord();
+      return parquetAvroSchema;
+    } catch (Exception e) {
+      LOG.error(String.format("Exception in converting avro schema to parquet format in ScribeParquetAvroConverter: Schema: %s, exception: %s", schema.getName(), e));
+      return null;
+    }
+
+  }
+
+  /**
+   * Gets the schema types from the field avro schema.
+   * @param fieldName  the field name
+   * @param avroRecord the incoming record
+   * @return
+   */
+  public static List<Schema> getSchemaTypes(String fieldName, GenericRecord avroRecord) {
+    try {
+      return ((GenericData.Record) avroRecord).getSchema().getField(fieldName).schema().isUnion() ?
+          ((GenericData.Record) avroRecord).getSchema().getField(fieldName).schema().getTypes() :
+          Arrays.asList(((GenericData.Record) avroRecord).getSchema().getField(fieldName).schema());
+    } catch (Exception e){
+      LOG.error(String.format("Exception in getting schema Types in ScribeParquetAvroConverter: field: %s, exception: %s", fieldName, e));
+      return null;
+    }
+  }
+
+  /**
+   * The incoming records need to be converted to match the Parquet schema generated using
+   * `generateParquetStructuredAvroSchema` function. This function iterates all the fields in the avroRecord {@link ScribeParquetAvroConverter#generateParquetStructuredAvroSchema(Schema)}
+   * and converts them to necessary data types to conform to the target Parquet schema.
+   *
+   * @param schema parquet compatible avro schema
+   * @param avroRecord the incoming record
+   * @return GenericRecord the record converted to match the new schema
+   * @throws Exception
```

Comment:
Should the method be declared as throws Exception? Also, why generic Exception object?